pkg/maps/lbmap/affinity.go

```diff
@@ -0,0 +1,180 @@
+	return &AffinityMatchKey{
+		BackendID: backendID,
+		RevNATID:  revNATID,
+	}
+}
+
+// GetKeyPtr returns the unsafe pointer to the BPF key
+func (k *AffinityMatchKey) GetKeyPtr() unsafe.Pointer { return unsafe.Pointer(k) }
+
+// GetValuePtr returns the unsafe pointer to the BPF value
+func (v *AffinityMatchValue) GetValuePtr() unsafe.Pointer { return unsafe.Pointer(v) }
+
+// String converts the key into a human readable string format
+func (k *AffinityMatchKey) String() string { return fmt.Sprintf("%d %d", k.BackendID, k.RevNATID) }
+
+// String converts the value into a human readable string format
+func (v *AffinityMatchValue) String() string { return "" }
+
+// NewValue returns a new empty instance of the structure representing the BPF
+// map value
+func (k *AffinityMatchKey) NewValue() bpf.MapValue { return &AffinityMatchValue{} }
+
+// ToNetwork returns the key in the network byte order
+func (k *AffinityMatchKey) ToNetwork() *AffinityMatchKey {
+	n := *k
+	// For some reasons rev_nat_index is stored in network byte order in
+	// the SVC BPF maps
+	n.RevNATID = byteorder.HostToNetwork(n.RevNATID).(uint16)
+	return &n
+}
+
+// Affinity4Key is the Go representation of lb4_affinity_key
+// +k8s:deepcopy-gen=true
+// +k8s:deepcopy-gen:interfaces=github.com/cilium/cilium/pkg/bpf.MapKey
+type Affinity4Key struct {
+	ClientID    uint64 `align:"client_id"`
+	RevNATID    uint16 `align:"rev_nat_id"`
+	NetNSCookie uint8  `align:"netns_cookie"`
+	Pad1        uint8  `align:"pad1"`
+	Pad2        uint32 `align:"pad2"`
+}
+
+// Affinity6Key is the Go representation of lb6_affinity_key
+// +k8s:deepcopy-gen=true
+// +k8s:deepcopy-gen:interfaces=github.com/cilium/cilium/pkg/bpf.MapKey
+type Affinity6Key struct {
+	ClientID    types.IPv6 `algin:"client_id"`
+	RevNATID    uint16     `align:"rev_nat_id"`
+	NetNSCookie uint8      `align:"netns_cookie"`
+	Pad         uint8      `align:"pad"`
+}
+
+// AffinityValue is the Go representing of lb_affinity_value
+// +k8s:deepcopy-gen=true
+// +k8s:deepcopy-gen:interfaces=github.com/cilium/cilium/pkg/bpf.MapValue
+type AffinityValue struct {
+	LastUsed  uint64 `align:"last_used"`
+	BackendID uint32 `align:"backend_id"`
+	Pad       uint32 `align:"pad"`
+}
+
+// GetKeyPtr returns the unsafe pointer to the BPF key
+func (k *Affinity4Key) GetKeyPtr() unsafe.Pointer { return unsafe.Pointer(k) }
+
+// GetKeyPtr returns the unsafe pointer to the BPF key
+func (k *Affinity6Key) GetKeyPtr() unsafe.Pointer { return unsafe.Pointer(k) }
+
+// GetValuePtr returns the unsafe pointer to the BPF value
+func (v *AffinityValue) GetValuePtr() unsafe.Pointer { return unsafe.Pointer(v) }
+
+// String converts the key into a human readable string format.
+func (k *Affinity4Key) String() string {
+	return fmt.Sprintf("%d %d %d", k.ClientID, k.NetNSCookie, k.RevNATID)
+}
+
+// String converts the key into a human readable string format.
+func (k *Affinity6Key) String() string { return "TODO" }
```

Comment:
Why TODO?

Reply:
Oops, slipped unnoticed. Will return the same val as `Affinity4Key.String()`.